source/contribute.rst

```diff
@@ -192,6 +195,70 @@ this in case you want to read more about this:
 - `Commit message style for git <https://commit.style/>`_
 - `A note about git commit messages <https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html>`_
 
+To squash or not to squash
+--------------------------
+
+Of course, if you want to craft good commit messages, you will have a
+hard time if the changeset you are describing does too many things. That
+might very well happen if you notice small things along the way that are
+unrelated to your PR, but too small to warrant a separate one. ``git add
+--patch`` or ``git add -p`` will be of invaluable help to commit things
+separately.
+On the contrary, there are commits that typically do not need to exist,
+such as commits that fix coding style or address minor review comments.
+Bear in mind that the git log is not only aimed at reviewers, but also
+at anyone who wants to understand some change you made. Do not distract
+them with cs fixes and try to make it look as if you produced perfect
```

Comment:
It took me three times to read it to make sure I understand it correctly :-) It could be interpreted as: "Do not distract […] and [do not] try to make it look […]".

I'd break it up into two sentences: Do not distract […]. Instead, try to make it look […].

Additionally, the "make it look" to me seems to carry some negative connotation as if we're asking the contributor to fake something. Instead, it could be worded as "make a commit that contains the changes you wanted to make _and_ you had to make on top of those because of the code style, reviews, etc" (the wording is to be adjusted, this is just about the delivery).